api/raft.cc

```diff
@@ -74,12 +74,32 @@ void set_raft(http_context&, httpd::routes& r, sharded<service::raft_group_regis
             return json_return_type(srv.current_leader().to_sstring());
         });
     });
+    r::read_barrier.set(r, [&raft_gr] (std::unique_ptr<http::request> req) -> future<json_return_type> {
+        auto srv = std::invoke([&] {
+            if (!req->query_parameters.contains("group_id")) {
+                return raft_gr.local().group0_with_timeouts();
+            }
+            raft::group_id gid{utils::UUID{req->get_query_param("group_id")}};
+            return raft_gr.local().get_server_with_timeouts(gid);
+        });
+        auto timeout = std::invoke([timeout_str = req->get_query_param("timeout")] {
+            if (timeout_str.empty()) {
+                return ::service::raft_timeout{};
+            }
+            auto dur = std::stoll(timeout_str);
+            if (dur <= 0) {
+                throw std::runtime_error{"Timeout must be a positive number."};
```

Comment:
I see that other code in api/ throws bad_param_exception when they encounter bad parameters, not the generic std::runtime_error. Please consider if this code should do it too.

Reply:
Yeah, this was copied from the other request slightly above, but `bad_param_exception` might make more sense.

Reply:
changed to `bad_param_exception`